modules/misc/xdg-mime-apps.nix

```diff
@@ -0,0 +1,104 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.xdg.mimeApps;
+
+  strListOrSingleton = with types; coercedTo
+    (either (listOf str) str)
+    (v: if isList v then v else [v])
+    (listOf str);
+
+  writeAssocs = let
+    writeAssoc = key: vals: "${key}=${concatStringsSep ";" vals}";
+  in assocs: concatStringsSep "\n" (mapAttrsToList writeAssoc assocs);
+
+in {
+  meta.maintainers = with maintainers; [ pacien ];
+
+  options.xdg.mimeApps = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to manage <filename>$XDG_CONFIG_HOME/mimeapps.list</filename>.
+        This option requires <option>xdg.enable</option> to be enabled.
+        </para>
+        <para>
+        The generated file is read-only.
```

Comment:
Using `mkOption` instead of `mkEnableOption` here because of the lengthy description.